sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowTablesExec.scala

```diff
@@ -33,15 +41,36 @@ case class ShowTablesExec(
     output: Seq[Attribute],
     catalog: TableCatalog,
     namespace: Seq[String],
-    pattern: Option[String]) extends V2CommandExec with LeafExecNode {
+    pattern: Option[String],
+    isExtended: Boolean = false,
+    partitionSpec: Option[ResolvedPartitionSpec] = None) extends V2CommandExec with LeafExecNode {
   override protected def run(): Seq[InternalRow] = {
     val rows = new ArrayBuffer[InternalRow]()
 
-    val tables = catalog.listTables(namespace.toArray)
-    tables.map { table =>
-      if (pattern.map(StringUtils.filterPattern(Seq(table.name()), _).nonEmpty).getOrElse(true)) {
-        rows += toCatalystRow(table.namespace().quoted, table.name(), isTempView(table))
+    if (partitionSpec.isEmpty) {
+      // Show the information of tables.
+      val identifiers = catalog.listTables(namespace.toArray)
+      identifiers.map { identifier =>
+        if (pattern.map(StringUtils.filterPattern(
+          Seq(identifier.name()), _).nonEmpty).getOrElse(true)) {
+          val isTemp = isTempView(identifier)
+          if (isExtended) {
+            val table = catalog.loadTable(identifier)
```

Comment:
seems we should not load the table if it's a temp view?

Reply:
From the current logic, it seems that if the catalog is V2SessionCatalog, it can be loaded, although we can see from the following code comments that a `NoSuchTableException` should be thrown at this point.
https://github.com/apache/spark/blob/2709426f0f622a214c51954664458e7dd2ab3304/sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableCatalog.java#L103-L104
If my understanding is incorrect, please correct me.